docs/source/data/errors.mdx

```diff
@@ -332,130 +346,135 @@ You can edit Apollo Server error details before they're passed to a client or re
+  // highlight-start
+  formatError: (formattedError, error) => {
+    // Return a different error message
+    if (
+      formattedError.extensions.code ===
+      ApolloServerErrorCode.GRAPHQL_VALIDATION_FAILED
+    ) {
+      return {
+        ...formattedError,
+        message: "Your query doesn't match the schema. Try double-checking it!",
+      };
     }
 
-    // Otherwise return the original error. The error can also
+    // Otherwise return the formatted error. This error can also
     // be manipulated in other ways, as long as it's returned.
-    return err;
+    return formattedError;
   },
+  // highlight-end
 });
 
-server.listen().then(({ url }) => {
-  console.log(`🚀 Server ready at ${url}`);
-});
+const { url } = await startStandaloneServer(server, { listen: { port: 4000 } });
+console.log(`🚀 Server listening at: ${url}`);
 ```
 
-The error instance received by `formatError` (a `GraphQLError`) contains an `originalError` property, which represents the original error thrown in the resolver. You can use this property to obtain the `instanceof` the error class, such as `AuthenticationError` or `ValidationError`:
+As another example, here we return a more generic error whenever the original error's message begins with `Database Error: `:
 
-```js
-  formatError(err) {
-    if (err.originalError instanceof AuthenticationError) {
-      return new Error('Different authentication error message!');
-    }
-  },
+```ts
+formatError: (formattedError, error) => {
+  // highlight-start
+  if (formattedError.message.startsWith('Database Error: ')) {
+    return { message: 'Internal server error' };
+  }
+  // highlight-end
+
+  // Otherwise return the formatted error.
+  return formattedError;
+},
 ```
 
-> To make context-specific adjustments to the error received by `formatError` (such as localization or personalization), consider [creating a plugin](../integrations/plugins/) that uses the [`didEncounterErrors` lifecycle event](../integrations/plugins-event-reference/#didencountererrors) to attach additional properties to the error. These properties can be accessed from `formatError`.
+> To make context-specific adjustments to the error received by `formatError` (such as localization or personalization), consider [creating a plugin](/apollo-server/integrations/plugins/) that uses the [`didEncounterErrors` lifecycle event](/apollo-server/integrations/plugins-event-reference/#didencountererrors) to attach additional properties to the error. These properties can be accessed from `formatError`.
 
 ### For Apollo Studio reporting
 
 You can use Apollo Studio to analyze your server's error rates. If you connect Apollo Server to Studio, all errors are sent to Studio by default. If you _don't_ want certain error information to be sent to Studio (either because the error is unimportant or because certain information is confidential), you can modify or redact errors entirely before they're transmitted.
```

Comment:
> all errors are sent to Studio by default

This is a bit of an exaggeration. It's more like "operations sent to Studio as detailed traces contain every error". Not that literally every error gets sent, because not all operations are sent to Studio as traces at all. In fact, basically none are if you're not a paid user (each time Apollo Server starts up it might send one message with traces and the response to that trace will tell it to stop sending traces if you're not paid).

(Also I mention below that as of ultra recently the default is to mask all errors)

Reply:
I love the phrasing you suggested, so is now:

> If you connect Apollo Server to Studio, operations sent to Studio as detailed traces contain every error by default.

Reply:
Wait, @glasser isn't this no longer true? Since the usage reporting plugin now omits [error details from traces by default](https://www.apollographql.com/docs/apollo-server/v4/migration#usage-reporting-and-inline-trace-plugins-mask-errors-by-default)?

Reply:
Yeah, that's what I said in the second parenthetical paragraph.

My point is twofold:

- The original wording was just misleading even in the context of AS3: it implies *every* error generated gets sent to Studio, which has never been true.
- Also we mask it now

Reply:
Sweet, thank you for the clarification!